plugin-openshift/che-plugin-openshift-ext-client/src/main/java/org/eclipse/che/ide/ext/openshift/client/delete/DeleteProjectPresenter.java

```diff
@@ -0,0 +1,179 @@
+        return new Operation<List<BuildConfig>>() {
+            @Override
+            public void apply(List<BuildConfig> configs) throws OperationException {
+                String dialogLabel = "";
+                if (configs.isEmpty()) {
+                    dialogLabel = locale.deleteProjectWithoutAppLabel(nameSpace);
+                }
+                if (configs.size() == 1) {
+                    dialogLabel = locale.deleteSingleAppProjectLabel(nameSpace);
+                }
+                if (configs.size() > 1) {
+                    String applications = getBuildConfigNames(configs);
+                    dialogLabel = locale.deleteMultipleAppProjectLabel(nameSpace, applications);
+                }
+                dialogFactory.createConfirmDialog(locale.deleteProjectDialogTitle(),
+                                                  dialogLabel,
+                                                  new ConfirmCallback() {
+                                                      @Override
+                                                      public void accepted() {
+                                                          service.deleteProject(nameSpace).then(new Operation<Void>() {
+                                                              @Override
+                                                              public void apply(Void arg) throws OperationException {
+                                                                  notificationManager.showInfo(locale.deleteNameSpaceSuccess(nameSpace));
+                                                                  cleanUpApplication(projectDescriptor, nameSpace);
+                                                              }
+                                                          }).catchError(new Operation<PromiseError>() {
+                                                              @Override
+                                                              public void apply(PromiseError arg) throws OperationException {
+                                                                  handleError(nameSpace);
+                                                              }
+                                                          });
+                                                      }
+                                                  },
+                                                  null).show();
+            }
+        };
+    }
+
+    private Operation<PromiseError> handleError(final String nameSpace) {
+        return new Operation<PromiseError>() {
+            @Override
+            public void apply(PromiseError promiseError) throws OperationException {
+                notificationManager.showError(locale.deleteProjectFailed(nameSpace) + " " + promiseError.getMessage());
+            }
+        };
+    }
+
+    private void cleanUpApplication(final ProjectDescriptor descriptor, final String nameSpace) {
```

Comment:
It is not good name for this method. I think it would be better to name it as removeOpenshiftMixins or something like this